tests/middleware_test.py

```diff
@@ -0,0 +1,222 @@
+import asyncio
+import json
+from types import MethodType
+from typing import AsyncIterator
+
+import anyio
+import httpx
+from mcp.types import (
+    LATEST_PROTOCOL_VERSION,
+    CallToolResult,
+    ClientCapabilities,
+    ClientNotification,
+    Implementation,
+    InitializeRequest,
+    InitializeRequestParams,
+    InitializedNotification,
+    JSONRPCNotification,
+    JSONRPCRequest,
+    JSONRPCResponse,
+)
+import pytest
+from httpx_sse import aconnect_sse
+
+from mcp_grafana import mcp
+from mcp_grafana.middleware import run_sse_async_with_middleware
+
+from pytest_httpserver import HTTPServer
+
+
+class TestMiddleware:
+    """
+    Test that our injected starlette middleware extracts headers and
+    overrides settings per-request, as expected.
+
+    Also ensure that the contextvars do not leak across requests.
+    """
+
+    @pytest.mark.asyncio
+    async def test_multiple_requests(self):
+        """
+        Ensure that the contextvars do not leak across requests.
+
+        This is a bit of a tricky test, since we're not actually
+        testing the middleware itself, but rather the contextvars
+        that it uses.
```

Comment:
Found this a bit confusing since we can't peek into the contextvars (IIUC) and we're using middlewares as a proxy for the test.

Reply:
Ugh that comment was added by some AI tool and isn't even true. I'll remove it!